### ISSUE TITLE
Hub 01 Mission 2, Trying to get equipment shared from the intercom after trading crashes the game

### ISSUE THREAD

**Issue by Elsharr** [CleverRaven/Cataclysm-DDA#40709]
# Describe the bug
When starting mission 2 for the intercom in Hub 01, if you trade first then try to request sharing equipment after making a trade, the game crashes if the request succeeds.

The error doesn't occur if you accept the mission and request equipment without trading first.  Otherwise the failure is repeatable, it happens every time I tried it in my save.

You may have to try requesting equipment several times, my barter/talk skill is terrible.  After several attempts, an Error box appears with:
The program has crashed.  See the log file for a stacktrace.  
CRASH LOG FILE: ./config/crash.log 
VERSION 0.E 
TYPE: Unexpected termination
MESSAGE: Unexpected Termination

# Steps To Reproduce
Based on my save, where mission 1 is already completed and turned in:
1.  Examine the intercom
2.  Trade (I traded 1 of their science ID badges for a towel and a dirty damaged fit-over sunglasses)
3. Start Mission 2, try to negotiate for shared equipment.
4.  Fail.
5.  Try again, repeat until crash.

# Expected behavior
I expected to either get some free equipment, annoy the voice on the intercom to the point where it refused to talk to me anymore, or some other bad thing like a trapdoor, laser, giant anvil out of the ceiling, etc.

# Screenshots
Only applicable screenshot is the conversation screen with the error log overlaid.  The template suggested not including a crash screenshot, so if you need it let me know.

# Versions and configuration
- OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.E [64-bit]
- Graphics Version: Tiles
- Game Language: System language [] (English)
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Stats Through Skills [StatsThroughSkills]
]

# Additional context
Included are the debug log, crash log, and copy of the savegame.  They're all in the one zip file.   I'm right next to the terminal and about to talk to it.   I've tried it a few more times to see if it's reproducible.

[info packet.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4652838/info.packet.zip)

**Comment by mlangsdorf** [comment]
I can't reproduce the crash, either from the save game or by attempting the replication steps myself.  If @Elsharr doesn't comment on this in a week or so, I'm going to close it as fixed.

**Comment by Elsharr** [comment]
I'm going to go through the process again.  I'm using the stable 0.E version, and I'll take screenshots at each step.  I just unzipped the support package and loaded it as my savegame.   I'll also add more comments to what I think the problem might be.

I'll upload a series of screenshots that correspond with the steps.

1. Examine the intercom
2. Trade (I traded 1 of their science ID badges for a towel and a dirty damaged fit-over sunglasses)
3. Start Mission 2, try to negotiate for shared equipment (trying Persuade for best odds).
4. Fail. (actually, it looks like I succeeded first try this time, game crashed on "success" first try)
5. Try again, repeat until crash. (didn't have to retry this time, straight to crash)

If you persuade without trading first, everything's fine.  If you trade and don't persuade, everything's fine.   I noticed that if you persuade, you still end up with debt for what they gave you.   Maybe it has something to do with having a little bit of credit (after the trade) and then dropping below zero?   In any case, for me, and on different computers, this set of steps with this save game consistently causes the game crash.   
[bugreport.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/5096429/bugreport.zip)
